config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java

```diff
@@ -33,13 +34,15 @@ public BundleMetadata build(final AnnotatedBundle annotatedBundle, final Annotat
         final Encass encass = (Encass) annotatedEntity.getEntity();
         final String bundleName = annotatedBundle.getBundleName();
         final String name = bundleName.substring(0, bundleName.indexOf(projectVersion) - 1);
+        final boolean isPolicyOrEncassReusable = annotatedBundle.getEntities(Policy.class).entrySet().stream().anyMatch(entity -> entity.getValue().isReusable()) ||
+                annotatedBundle.getEntities(Encass.class).entrySet().stream().anyMatch(entity -> entity.getValue().isReusable());
```

Comment:
You can mode this logic to a different method. And then bellow, you can do like this:
`builder.reusableAndRedeployable(true, annotatedEntity.isRedeployable() || !bundleContainsReusableEntity());`
This way we avoid iterating the policies and encasses if `@redeployable` is set.

Reply:
done.